--- e2fsprogs-1.39/lib/blkid/probe.c.fatlabel	2006-09-17 15:45:54.000000000 +0200
+++ e2fsprogs-1.39/lib/blkid/probe.c	2006-09-17 15:46:32.000000000 +0200
@@ -348,8 +348,8 @@
 	}
 
 	if (vol_label && memcmp(vol_label, no_name, 11)) {
-		label = vol_label;
-		label_len = figure_label_len(vol_label, 11);
+		if ((label_len = figure_label_len(vol_label, 11)))
+			label = vol_label;
 	}
 
 	/* We can't just print them as %04X, because they are unaligned */
--- e2fsprogs-1.39/lib/blkid/ChangeLog.fatlabel	2006-09-17 15:51:05.000000000 +0200
+++ e2fsprogs-1.39/lib/blkid/ChangeLog	2006-09-17 15:52:37.000000000 +0200
@@ -0,0 +1,4 @@
+2006-09-17  Karel Zak  <kzak@redhat.com> 
+
+	* probe.c (probe_fat): Fix problem with empty FAT label.
+
