Index: e2fsprogs/doc/libext2fs.texinfo
===================================================================
--- e2fsprogs.orig/doc/libext2fs.texinfo	2009-01-28 20:56:26.000000000 -0600
+++ e2fsprogs/doc/libext2fs.texinfo	2009-01-29 16:36:53.148590911 -0600
@@ -8,7 +8,7 @@
 @ifinfo
 @dircategory Development
 @direntry
-* libext2fs: (libext2fs.info).                  The EXT2FS library.
+* libext2fs: (libext2fs).                  The EXT2FS library.
 @end direntry
 @end ifinfo
 
