

diff --git a/lib/support/mkquota.c b/lib/support/mkquota.c
index 00f3a40..47b702d 100644
--- a/lib/support/mkquota.c
+++ b/lib/support/mkquota.c
@@ -50,7 +50,7 @@ static void print_dquot(const char *desc, struct dquot *dq)
 {
 	if (desc)
 		fprintf(stderr, "%s: ", desc);
-	fprintf(stderr, "%u %ld:%ld:%ld %ld:%ld:%ld\n",
+	fprintf(stderr, "%u %lld:%lld:%lld %lld:%lld:%lld\n",
 		dq->dq_id, dq->dq_dqb.dqb_curspace,
 		dq->dq_dqb.dqb_bsoftlimit, dq->dq_dqb.dqb_bhardlimit,
 		dq->dq_dqb.dqb_curinodes,
@@ -524,7 +524,7 @@ static int scan_dquots_callback(struct dquot *dquot, void *cb_data)
 	    dq->dq_dqb.dqb_curinodes != dquot->dq_dqb.dqb_curinodes) {
 		scan_data->usage_is_inconsistent = 1;
 		fprintf(stderr, "[QUOTA WARNING] Usage inconsistent for ID %u:"
-			"actual (%ld, %ld) != expected (%ld, %ld)\n",
+			"actual (%lld, %lld) != expected (%lld, %lld)\n",
 			dq->dq_id, dq->dq_dqb.dqb_curspace,
 			dq->dq_dqb.dqb_curinodes,
 			dquot->dq_dqb.dqb_curspace,
diff --git a/lib/support/quotaio.c b/lib/support/quotaio.c
index 2daf178..ef63056 100644
--- a/lib/support/quotaio.c
+++ b/lib/support/quotaio.c
@@ -230,9 +230,9 @@ errcode_t quota_file_open(quota_ctx_t qctx, struct quota_handle *h,
 		return err;
 
 	if (qf_ino == 0)
-		qf_ino = *quota_sb_inump(fs->super, qtype);
+		qf_ino = *quota_sb_inump(fs->super, qtype)
 
-	log_debug("Opening quota ino=%u, type=%d", qf_ino, qtype);
+	log_debug("Opening quota ino=%lu, type=%d", qf_ino, qtype);
 	err = ext2fs_file_open(fs, qf_ino, flags, &e2_file);
 	if (err) {
 		log_err("ext2fs_file_open failed: %s", error_message(err));
@@ -368,7 +368,7 @@ errcode_t quota_file_create(struct quota_handle *h, ext2_filsys fs,
 	h->e2fs_write = quota_write_nomount;
 	h->e2fs_read = quota_read_nomount;
 
-	log_debug("Creating quota ino=%u, type=%d", qf_inum, qtype);
+	log_debug("Creating quota ino=%lu, type=%d", qf_inum, type);
 	err = ext2fs_file_open(fs, qf_inum, h->qh_file_flags, &e2_file);
 	if (err) {
 		log_err("ext2fs_file_open failed: %ld", err);


