Index: e2fsprogs-1.41.5/doc/libblkid.txt
===================================================================
--- e2fsprogs-1.41.5.orig/doc/libblkid.txt
+++ e2fsprogs-1.41.5/doc/libblkid.txt
@@ -58,7 +58,7 @@ type, or label, or uuid for a particular
 	}
 
 If a program needs to call multiple blkid functions, then passing in a
-cache value of NULL is not recommended, since the /etc/blkid.tab file
+cache value of NULL is not recommended, since the /etc/blkid/blkid.tab file
 will be repeatedly parsed over and over again, with memory allocated
 and deallocated.  To initialize the blkid cache, blkid_get_cache()
 function is used:
@@ -68,7 +68,7 @@ function is used:
 
 The second parameter of blkid_get_cache (if non-zero) is the alternate
 filename of the blkid cache file (where the default is
-/etc/blkid.tab).  Normally, programs should just pass in NULL.
+/etc/blkid/blkid.tab).  Normally, programs should just pass in NULL.
 
 If you have called blkid_get_cache(), you should call blkid_put_cache()
 when you are done using the blkid library functions.  This will save the
Index: e2fsprogs-1.41.5/misc/blkid.8.in
===================================================================
--- e2fsprogs-1.41.5.orig/misc/blkid.8.in
+++ e2fsprogs-1.41.5/misc/blkid.8.in
@@ -55,7 +55,7 @@ more devices.
 Read from
 .I cachefile
 instead of reading from the default cache file
-.IR /etc/blkid.tab .
+.IR /etc/blkid/blkid.tab .
 If you want to start with a clean cache (i.e. don't report devices previously
 scanned but not necessarily available at this time), specify
 .IR /dev/null .
@@ -133,7 +133,7 @@ Display version number and exit.
 Write the device cache to
 .I writecachefile
 instead of writing it to the default cache file
-.IR /etc/blkid.tab .
+.IR /etc/blkid/blkid.tab .
 If you don't want to save the cache to the default file, specify
 .IR /dev/null.
 If not specified it will be the same file as that given by the
Index: e2fsprogs-1.41.5/misc/blkid.c
===================================================================
--- e2fsprogs-1.41.5.orig/misc/blkid.c
+++ e2fsprogs-1.41.5/misc/blkid.c
@@ -52,7 +52,7 @@ static void usage(int error)
 	fprintf(out,
 		"usage:\t%s [-c <file>] [-ghlLv] [-o format] "
 		"[-s <tag>] [-t <token>]\n    [-w <file>] [dev ...]\n"
-		"\t-c\tcache file (default: /etc/blkid.tab, /dev/null = none)\n"
+		"\t-c\tcache file (default: /etc/blkid/blkid.tab, /dev/null = none)\n"
 		"\t-h\tprint this usage message and exit\n"
 		"\t-g\tgarbage collect the blkid cache\n"
 		"\t-s\tshow specified tag(s) (default show all tags)\n"
Index: e2fsprogs-1.41.5/lib/blkid/Makefile.in
===================================================================
--- e2fsprogs-1.41.5.orig/lib/blkid/Makefile.in
+++ e2fsprogs-1.41.5/lib/blkid/Makefile.in
@@ -142,9 +142,10 @@ blkid.pc: $(srcdir)/blkid.pc.in $(top_bu
 	@cd $(top_builddir); CONFIG_FILES=lib/blkid/blkid.pc ./config.status
 
 installdirs::
-	@echo "	MKINSTALLDIRS $(libdir) $(includedir)/blkid"
+	@echo "	MKINSTALLDIRS $(libdir) $(includedir)/blkid /etc/blkid"
 	@$(MKINSTALLDIRS) $(DESTDIR)$(libdir) \
-		$(DESTDIR)$(includedir)/blkid $(DESTDIR)$(libdir)/pkgconfig
+		$(DESTDIR)$(includedir)/blkid $(DESTDIR)$(libdir)/pkgconfig \
+		$(DESTDIR)/etc/blkid
 
 install:: all installdirs 
 	@echo "	INSTALL_DATA $(libdir)/libblkid.a"
@@ -170,6 +171,7 @@ uninstall::
 	$(RM) -f $(DESTDIR)$(libdir)/libblkid.a \
 		$(DESTDIR)$(libdir)/pkgconfig/blkid.pc
 	$(RM) -rf $(DESTDIR)$(includedir)/blkid
+	$(RM) -rf $(DESTDIR)/etc/blkid
 	for i in $(SMANPAGES); do \
 		$(RM) -f $(DESTDIR)$(man3dir)/$$i; \
 	done
Index: e2fsprogs-1.41.5/lib/blkid/libblkid.3.in
===================================================================
--- e2fsprogs-1.41.5.orig/lib/blkid/libblkid.3.in
+++ e2fsprogs-1.41.5/lib/blkid/libblkid.3.in
@@ -24,7 +24,7 @@ A common use is to allow use of LABEL= a
 specific block device names into configuration files.
 .P
 Block device information is normally kept in a cache file
-.I /etc/blkid.tab
+.I /etc/blkid/blkid.tab
 and is verified to still be valid before being returned to the user
 (if the user has read permission on the raw block device, otherwise not).
 The cache file also allows unprivileged users (normally anyone other
@@ -51,7 +51,7 @@ was written by Andreas Dilger for the ex
 from Ted Ts'o.  The library was subsequently heavily modified by Ted Ts'o.
 .SH FILES
 .TP
-.I /etc/blkid.tab
+.I /etc/blkid/blkid.tab
 Caches data extracted from each recognized block device.
 .SH AVAILABILITY
 .B libblkid
Index: e2fsprogs-1.41.5/lib/blkid/blkidP.h
===================================================================
--- e2fsprogs-1.41.5.orig/lib/blkid/blkidP.h
+++ e2fsprogs-1.41.5/lib/blkid/blkidP.h
@@ -104,7 +104,7 @@ struct blkid_struct_cache
 extern char *blkid_strdup(const char *s);
 extern char *blkid_strndup(const char *s, const int length);
 
-#define BLKID_CACHE_FILE "/etc/blkid.tab"
+#define BLKID_CACHE_FILE "/etc/blkid/blkid.tab"
 
 #define BLKID_ERR_IO	 5
 #define BLKID_ERR_PROC	 9
Index: e2fsprogs-1.41.5/RELEASE-NOTES
===================================================================
--- e2fsprogs-1.41.5.orig/RELEASE-NOTES
+++ e2fsprogs-1.41.5/RELEASE-NOTES
@@ -2829,7 +2829,7 @@ label.
 
 Fixed a bug in the blkid library wihch could avoid an infinite loop
 in blkid_find_dev_with_tag() if /proc is not mounted and there the
-/etc/blkid.tab file has not yet been created.
+/etc/blkid/blkid.tab file has not yet been created.
 
 Fixed the badblocks program so that the destructive read/write test
 honors the -c option, and to use O_DIRECT when possible to avoid
