It looks like some debugging crept in?

Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Index: e2fsprogs-1.41.8/misc/filefrag.c
===================================================================
--- e2fsprogs-1.41.8.orig/misc/filefrag.c
+++ e2fsprogs-1.41.8/misc/filefrag.c
@@ -95,8 +95,6 @@ static int get_bmap(int fd, unsigned lon
 	int	ret;
 	unsigned int b;
 
-	printf("Calling get_bmap for block %lu\n", block);
-	abort();
 	b = block;
 	ret = ioctl(fd, FIBMAP, &b); /* FIBMAP takes pointer to integer */
 	if (ret < 0) {
