--- e2fsprogs-1.39/lib/blkid/probe.c.leak	2006-09-05 15:41:13.000000000 -0400
+++ e2fsprogs-1.39/lib/blkid/probe.c	2006-09-05 15:40:19.000000000 -0400
@@ -884,7 +884,12 @@
 	}
 
 	if (!dev->bid_type) {
-		if (probe.fd >= 0) close(probe.fd);
+		if (probe.fd >= 0)
+                    close(probe.fd);
+	        if (probe.sbbuf)
+		    free(probe.sbbuf);
+	        if (probe.buf)
+		    free(probe.buf);
 		blkid_free_dev(dev);
 		return NULL;
 	}
